packages/eas-cli/src/update/configure.ts

```diff
@@ -0,0 +1,294 @@
+  ) {
+    androidRuntimeVersion = androidRuntimeVersion ?? getDefaultRuntimeVersion(workflows.android);
+    iosRuntimeVersion = iosRuntimeVersion ?? getDefaultRuntimeVersion(workflows.ios);
+
+    if (platform === 'all' && isRuntimeEqual(androidRuntimeVersion, iosRuntimeVersion)) {
+      modifyConfig.runtimeVersion = androidRuntimeVersion;
+    } else {
+      if (['all', 'android'].includes(platform)) {
+        modifyConfig.runtimeVersion = undefined;
+        modifyConfig.android = { runtimeVersion: androidRuntimeVersion };
+      }
+      if (['all', 'ios'].includes(platform)) {
+        modifyConfig.runtimeVersion = undefined;
+        modifyConfig.ios = { runtimeVersion: iosRuntimeVersion };
+      }
+    }
+  }
+
+  if (Object.keys(modifyConfig).length === 0) {
+    return { exp, projectChanged: false };
+  }
+
+  // NOTE(cedric): might be better with a mergeDeep method, or handle in `modifyConfigAsync`
+  const mergedExp = {
+    ...exp,
+    runtimeVersion: modifyConfig.runtimeVersion ?? exp.runtimeVersion,
+    updates: { ...exp.updates, url: modifyConfig.updates?.url ?? exp.updates?.url },
+    android: {
+      ...exp.android,
+      runtimeVersion: modifyConfig.android?.runtimeVersion ?? exp.android?.runtimeVersion,
+    },
+    ios: {
+      ...exp.ios,
+      runtimeVersion: modifyConfig.ios?.runtimeVersion ?? exp.ios?.runtimeVersion,
+    },
+  };
+
+  // TODO(cedric): check where these properties are coming from, they pop up in `eas update`
+  delete mergedExp.sdkVersion; // Note(cedric): this is always returned from `getConfig`, but we don't want to set it
+  delete mergedExp['_internal'];
+  delete mergedExp.originalFullName;
+  delete mergedExp.currentFullName;
+  delete mergedExp.platforms;
+  const result = await modifyConfigAsync(projectDir, mergedExp);
+
+  switch (result.type) {
+    case 'success':
+      logEasUpdatesAutoConfig({ exp, modifyConfig });
+      return {
+        projectChanged: true,
+        // TODO(cedric): fix return type of `modifyConfigAsync` to avoid `null` for type === success repsonses
+        exp: result.config?.expo ?? mergedExp,
+      };
+
+    case 'warn':
+      warnEASUpdatesManualConfig({ modifyConfig, workflows });
+      throw new Error(result.message);
+
+    case 'fail':
+      throw new Error(result.message);
+
+    default:
+      throw new Error(
+        `Unexpected result type "${result.type}" received when modifying the project config.`
+      );
+  }
+}
+
+function logEasUpdatesAutoConfig({
+  modifyConfig,
+  exp,
+}: {
+  modifyConfig: Partial<ExpoConfig>;
+  exp: ExpoConfig;
+}): void {
+  if (modifyConfig.updates?.url) {
+    Log.withTick(
+      exp.updates?.url
+        ? `Overwrote updates.url "${exp.updates.url}" with "${modifyConfig.updates.url}"`
+        : `Configured updates.url to "${modifyConfig.updates.url}"`
+    );
+  }
+
+  if (modifyConfig.android?.runtimeVersion ?? modifyConfig.runtimeVersion) {
```

Comment:
not necessary but this block and the next (ios, android) should be a loop.

Reply:
Yeah, to do this, we need to simplify the types. Right now `appPlatformDisplayNames` uses `AppPlatform` (which is all caps `ANDROID`/`IOS`). But, `ExpoConfig` uses lowercase `Platform`.

So to minimize the type of juggling, I kept it the same.